### ISSUE TITLE
Reclaim vertical space on puzzle page

### ISSUE THREAD

**Issue by ebroder** [deathandmayhem/jolly-roger#1388]
Someone asked if it would be possible to collapse the puzzle header on the puzzle page to create more vertical space.

Thinking about it more, I alternatively wonder if we could stash most of that information into the navbar instead (though we'd need a strategy for overflow)

**Comment by flipdog** [comment]
Okay so I've been playing around with this a bit. The current layout is like so:

![current-puzzle-layout](https://user-images.githubusercontent.com/2754711/213569530-8e2e95e1-2184-471a-af5c-03425481f0ea.JPG)

I think you could logically move some stuff up to the navbar with the breadcrumbs like the puzzle out-link and the "Edit" button.

Putting the guess button and correct answers adjacent to each other also feels logical. Then I think you could move up the tags to be in a row with the guess/answers, and keep it from overflowing in unwieldly ways by just limiting the number of tags that show and having the rest be a drop-down on hover. And let's face it, when you're on the puzzle page, immediate visibility of most of the tags is not actually helpful information to surface.

![puzzle-layout-mockup](https://user-images.githubusercontent.com/2754711/213569987-04ddfb86-5bba-4d12-b839-ff8fb9ea7d8c.png)

That would earn you back two full lines, and still leave room for multiple correct answers or very long answers if they come up.

**Comment by grantaelliott** [comment]
_For reference, for most unsolved puzzles, the current metadata pane occupies 76 vertical pixels with two rows of content. This increases once the puzzle is solved or if the tag list is long enough to wrap._

I'm weakly onboard with making the metadata pane collapsible - for most, that's probably a better workspace after an initial review of the puzzle. It adds an extra click to opening the puzzle, guessing, and inserting an image (which don't happen frequently). Arguably it makes solves less obvious, but we have the chat announcement. If we do this, we should also add chat announcements for priority changes, since those tags could now be out of sight. There's some discussion to be had regarding where the show button goes.

I think there's also a case for just making tags collapsible, which would reduce the metadata pane of an unsolved puzzle to a single row (with an extra button for hide/show tags) below the navbar instead of the (typical) two.

I'm strongly opposed to moving metadata contents into the navbar. Most importantly, it violates user assumptions about what the navbar is - action buttons and external links strike me as particularly egregious in this regard. More practically, the navbar already has to wrap to avoid truncating longer titles at common viewport widths and maximally preserving given information should be a design goal. I think the proposed layout would frequently run into problems at portrait iPad or half-laptop widths and would likely have to fall back to the current approach at mobile widths. Finally, I think it's inevitable that we'll add features in the future and every new link or button will be a struggle in a layout based on an absolute upper bound in form factor.

(Separately, the mockup fits the answers and tags on a single row, which accounts for one of the two lines saved. Remember that the answer row occupies no vertical space when there are no answers - which, aside from multi-answer puzzles, is when players are using the sheet - so in most usage, this doesn't actually save any vertical lines, aside from the suggestion to overflow long tag lists with on-hover popover. That aspect is related to #288, which proved tricky because presenting only part of list is potentially misleading and nesting popovers is awkward. I think a global hide/show is actually better.)

**TLDR:** I understand the desire to reclaim vertical space, but I believe compressing the metadata pane or fitting parts of it into the navbar would be a regression in usability. I weakly support making the metadata pane or tag list hideable instead, which introduces a small amount of friction to some actions (for users who choose to hide it) but allows us to focus on clear presentation instead of density.

**Comment by zarvox** [comment]
My intuition broadly matches Grant's here -- I will argue pretty hard against overloading the navbar with a bunch of additional content.  While I think I could still *possibly* be convinced to put the puzzle link in the navbar next to the puzzle title, I think we'd have to remove the text label, which would probably harm discoverability and lose the information scent that a Puzzle link is separate from what jolly-roger is showing the user at the moment, and then also the click target would be really close to the other nav links at mobile widths in a space that we already have to be careful about with overflow-ellipsis.  In a similar vein, the edit button could (at a stretch) be not completely out of place logically in the same context, but practically the layout would be rough.  The rest of the action buttons definitely don't belong up in the navbar, from an information architecture perspective.

I think we can probably be smarter about showing/hiding the row of tags (and possibly answers).  As a strawman: we could make the metadata section for an unsolved puzzle take up only one row of vertical space for layout, and show the top row of the most important items that need to be easily found at all times (puzzle link, guess button), and then we could show the rest of the contents of metadata pane when the user mouses over or focuses on the metadata section.  I'm imagining this working in such a way that the size of the iframe remains constant and the expanded metadata section covers the full width of the top of the spreadsheet/iframe when expanded.

I suspect we'll want to continue to show puzzle answers by default; most puzzles have only one answer, and I think it's a good idea to have that answer be clearly visible on the page to make it obvious that the puzzle is solved, rather than requiring the user to mouse over the metadata pane to see that the puzzle is solved.  Today, the reflow from the answer appearing is a change that affects the layout of the sheet, so it's quite a visible change and harder to miss, which seems like a feature.

**Comment by zarvox** [comment]
We added collapse buttons (with automatic re-expansion under relevant conditions) for both the chat pane and the metadata pane in #2680, so I think I'm gonna call this closed.
